source/framework/analysis/src/TRestExpressionEvaluationProcess.cxx

```diff
@@ -0,0 +1,221 @@
+///   </expressionset>
+/// </addProcess>
+/// ```
+///
+/// The main difference to ROOT cut strings are the two facts that 1. the strings are not limited to
+/// boolean expressions and 2. that the boolean operatiors `&&` and `||` are instead refered to
+/// `and` and `or` respectively. The latter is due to the fact that `&` is an invalid character in
+/// XML and thus parsing them is broken (or the user has to write `&amp;`, which is kind of unaccptable).
+///
+/// Each expression is stored as REST metadata in form of a lisp-like representation of
+/// the input expression. For example the last `boolComplicated` example is represented as:
+/// ```
+/// (|| (< (/ hitsAna_energy 1000.0)) (&& (== tckAna_nTracks_X 1) (== tckAna_nTracks_Y 1)))
+/// ```
+///
+/// *An importang note*: Due to the fact that this depends on *runtime evaluation* of a binary
+/// tree, which stores the expression there is a performance penalty over native code!
+/// This means expressions should be rewritten as a native REST process in case they are
+/// generally useful and not specific.
+///
+///--------------------------------------------------------------------------
+///
+/// RESTsoft - Software for Rare Event Searches with TPCs
+///
+/// History of developments:
+///
+/// 2021-Mar: Base implementation based on `expressionEval` library
+///             Sebastian Schmidt
+///
+/// \class      TRestExpressionEvaluationProcess
+/// \author     Sebastian Schmidt (@Vindaar)
+///
+/// <hr>
+///
+
+#include "TRestExpressionEvaluationProcess.h"
+
+ClassImp(TRestExpressionEvaluationProcess);
+
+///////////////////////////////////////////////
+/// \brief Default constructor
+///
+TRestExpressionEvaluationProcess::TRestExpressionEvaluationProcess() { Initialize(); }
+
+///////////////////////////////////////////////
+/// \brief Function to initialize input/output event members and define the
+/// section name
+///
+/// In this case we re-use the initialization of TRestRawToSignalProcess
+/// interface class.
+///
+void TRestExpressionEvaluationProcess::Initialize() {
+    SetSectionName(this->ClassName());
+
+    fEvent = NULL;
+}
+
+///////////////////////////////////////////////
+/// \brief Process initialization.
+///
+/// TODO It should fill the fEventIDs vector.
+///
+void TRestExpressionEvaluationProcess::InitProcess() {
+    /// TOBE implemented
+}
+
+///////////////////////////////////////////////
+/// \brief The main processing event function
+///
+TRestEvent* TRestExpressionEvaluationProcess::ProcessEvent(TRestEvent* eventInput) {
+    fEvent = eventInput;
+
+    // TOBE implemented. If the event ID fEvent->GetID() is not in the list
+    // we should return NULL;
+    for(auto exprPair : fExprMap) {
+        auto value = evaluate(exprPair.second);
+	if(!value.isLeft()){
+	  fAnalysisTree->SetObservableValue((string)exprPair.first, (double)value.getRight());
```

Comment:
I would prefer a call to `SetObservableValue` so that all observable names follow the same convention, where we have `pcsName_obsName`.